docs/rules/no-private-properties.md

```diff
@@ -0,0 +1,40 @@
+# Detects usages of non-public property bindings (no-private-properties)
+
+It can be helpful to enforce naming conventions for `private` (and sometimes `protected`) members of an object. For example, prefixing private properties with a `__` allows them to be easily discerned when being inspected by tools that do not have knowledge of TypeScript (such as most debuggers).
+
+## Rule Details
+
+This rule allows you to enforce conventions for property names by their visibility. By default, it enforces nothing.
```

Comment:
We should reword this to make it clear we're not testing property names in general but rather disallowing non-public ones from being set.

Maybe:

> This rule allows you to enforce that all properties being set on elements are public.